src/compiler/crystal/compiler.cr

```diff
@@ -226,6 +226,24 @@ module Crystal
       Result.new program, node
     end
 
+    # :ditto:
+    #
+    # Yields a `Program` instance before compiling.
+    def compile_configure_program(source : Source | Array(Source), output_filename : String, & : Program -> Nil) : Result
```

Comment:
Needed a way to expose the program _before_ the source is parsed in order to set the callbacks on it. Not my favorite, but seems to work for now...

Reply:
I suppose `#compile` should call this method with an empty block instead of duplicating the implementation?